docs/docs/concepts/memory.md

```diff
@@ -303,8 +285,24 @@ def update_instructions(state: State, store: BaseStore):
 
 ![](img/memory/update-instructions.png)
 
-#### Few-shot examples
+## Writing memories
+
+While [humans often form long-term memories during sleep](https://medicine.yale.edu/news-article/sleeps-crucial-role-in-preserving-memory/), AI agents need a different approach. When and how should agents create new memories? There are at least two primary methods for agents to write memories: "on the hot path" and "in the background".
+
+![](img/memory/hot_path_vs_background.png)
+
+### Writing memories in the hot path
+
+Creating memories during runtime offers both advantages and challenges. On the positive side, this approach allows for real-time updates, making new memories immediately available for use in subsequent interactions. It also enables transparency, as users can be notified when memories are created and stored.
+
+However, this method also presents challenges. It increases complexity, as the agent must decide what to commit to memory alongside its primary task. This can impact performance, potentially slowing down response times due to additional decision-making and degrading tool-calling performance and task completion rates. Additionally, the multitasking involved may result in fewer memories being saved, affecting recall in future conversations.
```

Comment:
1. "performance" is generic (sounds like it might be quality) do we mean latency here?
2. Why is there a degradation of tool calling performance or task completion rate? (unclear to me)

Reply:
This is specifically assuming that memory-creation is a tool. I will add that for clarity.